#!/usr/bin/env python
PACKAGE = "franka_interactive_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness_x", double_t, 0, "Cartesian translational stiffness x", 850, 0, 1500)
gen.add("translational_stiffness_y", double_t, 0, "Cartesian translational stiffness y", 850, 0, 1500)
gen.add("translational_stiffness_z", double_t, 0, "Cartesian translational stiffness z", 850, 0, 1500)

gen.add("rotational_stiffness_x", double_t, 0, "Cartesian rotational stiffness x", 30, 0, 50)
gen.add("rotational_stiffness_y", double_t, 0, "Cartesian rotational stiffness y", 30, 0, 50)
gen.add("rotational_stiffness_z", double_t, 0, "Cartesian rotational stiffness z", 30, 0, 50)

# gen.add("all_TRANSLATIONAL_stiffness", double_t, 0, "Cartesian TRANSLATIONAL stiffness", 200, 0, 400)
# gen.add("all_ROTATIONAL_stiffness", double_t, 0, "Cartesian ROTATIONAL stiffness", 10, 0, 30)

gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0.5, 0, 100)
# gen.add("activate_tool_compensation", bool_t, 0, "trigger to activate external tool compensation (the desired compensation forces defined in ./config)", True)

exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_full_param"))
